src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/internal/FSPath.kt

```diff
@@ -35,9 +36,21 @@ class FSPath(val path: Path, val fs: FileSystem) {
     fun resolve(child: String): FSPath =
         FSPath(path.resolve(child), fs)
 
+    operator fun div(child: String): FSPath =
+        this.resolve(child)
+
     fun source(): Source =
         fs.source(path)
 
+    fun readUtf8(): String =
+        source().buffer().readUtf8()
+
+    /**
+     * Returns `null` if the file doesn't exist.
+     */
+    fun readUtf8OrNull(): String? =
+        if (this.exists) this.source().buffer().readUtf8() else null
```

Comment:
note: we could call newly created method `readUtf8` instead of duplicating the code here

Reply:
https://github.com/krzema12/snakeyaml-engine-kmp/pull/372/commits/0bb98cac5aa21ca07e27049f71c33cbaa0789950